### PR TITLE
Open Upgrade Django to 4.2 LTS

### PR DESCRIPTION
**Pull request by zubairshakoorarbisoft**
### Description
Issues: 
- https://github.com/openedx/enterprise-subsidy/issues/151
- https://github.com/openedx/enterprise-subsidy/issues/152

### Testing instructions

Add some, if applicable

### Merge checklist
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

### CODE REVIEW

File: requirements/constraints.txt [hunk 1]
```diff
@@ -8,7 +8,9 @@
 # pin when possible.  Writing an issue against the offending project and
 # linking to it here is good.
 
+Django<4.3
```
Comment (iamsobanjaved) [inline requirements/constraints.txt:11]: Add this constraint below the common constraint pin in this file.

**Review by iamsobanjaved — COMMENTED** [review]
Just one change

File: Makefile [hunk 2]
```diff
@@ -110,7 +114,7 @@ $(COMMON_CONSTRAINTS_TXT):
 
 upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
 upgrade: $(COMMON_CONSTRAINTS_TXT) ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
-	sed 's/django-simple-history==3.0.0//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
+	sed 's/Django<4.0//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
```
Comment (macdiesel) [inline Makefile:117]: `sed -i`  here too?

Comment (iamsobanjaved) [reply Makefile:117]: Will be fixed in a new PR

Comment (iamsobanjaved) [reply Makefile]: Fixed

**Comment by iamsobanjaved** [comment]
Related internal PR: 
- https://github.com/edx/edx-internal/pull/9350

**Review by macdiesel — APPROVED** [review]

**Pull request merged**